features/karaf/framework/src/main/resources/resources/bin/oh2_dir_layout

```diff
@@ -3,35 +3,41 @@
 # DIRNAME is the directory of karaf, setenv, etc.
 (unset CDPATH) >/dev/null 2>&1 && unset CDPATH
 
-export OPENHAB_HOME=`cd "$DIRNAME/.."; pwd`
+if [ -z ${OSH_HOME} ]; then
+    export OSH_HOME=`cd "$DIRNAME/.."; pwd`
+fi
 
-if [ -z ${OPENHAB_CONF} ]; then
-    export OPENHAB_CONF="${OPENHAB_HOME}/config"
+if [ -z ${OSH_CONF} ]; then
+    export OSH_CONF="${OSH_HOME}/config"
 fi
 
-if [ -z ${OPENHAB_RUNTIME} ]; then
-    export OPENHAB_RUNTIME="${OPENHAB_HOME}/data"
+if [ -z ${OSH_RUNTIME} ]; then
+    export OSH_RUNTIME="${OSH_HOME}/data"
 fi
 
-if [ -z ${OPENHAB_USERDATA} ]; then
-    export OPENHAB_USERDATA="${OPENHAB_HOME}/data"
+if [ -z ${OSH_USERDATA} ]; then
+    export OSH_USERDATA="${OSH_HOME}/data"
```

Comment:
I believe this one should or could be amended too. The userdata folder contains several important things such as `jsondb` with definition of all things and so on. Whole point of changing packaging was to keep userdata intact of `./karaf clean` (or `./start clean`) calls which truncate `data` directory.

Reply:
So you mean move it out from under OSH_HOME so that it is retained following an update? I'm not sure how to resolve that though - if it's not relative to the home folder, then we can't provide a default and the only option would be to not start if the user didn't provide this.

I think that the best approach is to leave it as it is so there is a default, but clearly the doc should recommend pointing it to somewhere outside of the runtime folder so that it remains persistent through rebuilds.

Reply:
Its fine to stat in home dir, just needs to be above runtime/data dir.

Reply:
Ok, so you mean move it to something like `OSH_HOME/userdata` - ie not in the `data` area?

Reply:
Yes, that's what I meant. Sorry for being unclear.